scripts/cfFindIP.sh

```diff
@@ -19,43 +19,64 @@
 
 set -o nounset                                  # Treat unset variables as an error
 
+# Check if 'parallel', 'nmap' and 'bc' packages are installed
+# If they are not, install them
+if [[ "$(uname)" == "Linux" ]]; then
+    command -v parallel >/dev/null 2>&1 || { echo >&2 "I require 'parallel' but it's not installed. Installing now..."; apt-get install -y parallel; }
+    command -v nmap >/dev/null 2>&1 || { echo >&2 "I require 'nmap' but it's not installed. Installing now..."; apt-get install -y nmap; }
+    command -v bc >/dev/null 2>&1 || { echo >&2 "I require 'bc' but it's not installed. Installing now..."; apt-get install -y bc; }
+elif [[ "$(uname)" == "Darwin" ]]; then
+    command -v parallel >/dev/null 2>&1 || { echo >&2 "I require 'parallel' but it's not installed. Installing now..."; brew install parallel; }
+    command -v nmap >/dev/null 2>&1 || { echo >&2 "I require 'nmap' but it's not installed. Installing now..."; brew install nmap; }
+    command -v bc >/dev/null 2>&1 || { echo >&2 "I require 'bc' but it's not installed. Installing now..."; brew install bc; }
+    command -v gtimeout >/dev/null 2>&1 || { echo >&2 "I require 'gtimeout' but it's not installed. Installing now..."; brew install coreutils; }
+fi
+
 threads="$1"
 
 cloudFlareIpList=$(curl -s -XGET https://www.cloudflare.com/ips-v4)
 now=$(date +"%Y%m%d-%H%M%S")
 scriptDir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-resultDir="$scriptDir/../result"
-resultFile="$resultDir/$now-result.cf"
-
-# Function fncCheckSubnet
-# Check Subnet
-function fncCheckSubnet {
-	local ipList resultFile 
-	ipList="$1"
-	resultFile="$2"
-	for ip in ${ipList}
-	do
-		if timeout 1 bash -c "</dev/tcp/$ip/443" > /dev/null 2>&1;
-		then
-			timeMil=$(timeout 2 curl -s -w '%{time_total}\n' --resolve scan.sudoer.net:443:"$ip" https://scan.sudoer.net/data.100K --output /dev/null | xargs -I {} echo "{} * 1000 /1" | bc )
-			if [[ "$timeMil" ]]
-			then
-				echo "OK $ip"
-				echo "$ip" >> "$resultFile"
-			fi
-		else
-			echo "FAILED $ip"
-		fi
-	done
+resultDir="$scriptDir"
```

Comment:
Please do not change the result directory
It supposed to be somewhere else in script root directory

Reply:
Revised in the latest commit, with the same folder structure just added a check to create the folder if it's not available to support the standalone execution of the script(in case not cloned from git with existing folders)

Reply:
Revised in the latest commit, with the same folder structure just added a check to create the folder if it's not available to support the standalone execution of the script(in case not cloned from git with existing folders)